File: Sample Apps/MedableTest/MedableTest/ViewController.m
//
//  ViewController.m
//  MedableTest
//
//  Copyright (c) 2015 Medable Inc. All rights reserved.
//

#import "ViewController.h"
#import <Medable/Medable.h>

@interface ViewController ()

@property (nonatomic, weak) IBOutlet UILabel *orgName;
@property (nonatomic, weak) IBOutlet UILabel *status;
@property (nonatomic, weak) IBOutlet UILabel *orgCode;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    self.orgName.text = @"";
    self.orgCode.text = @"";
    
    MDAPIClient *client = [MDAPIClient sharedClient];
    
    if ( client != nil )
    {
        self.status.text = @"Initialized";
    }
    else
    {
        self.status.text = @"Error on Init";
    }
}

-(IBAction)getOrganization:(id)sender
{
    MDAPIClient *client = [MDAPIClient sharedClient];
    
    [client getPublicOrgInfoWithCallback:^(NSDictionary *orgInfo, MDFault *fault)
    {
        if ( orgInfo != nil )
        {
            self.orgName.text = orgInfo[@"name"];
            self.orgCode.text = orgInfo[@"code"];
            self.status.text = @"Ok";
        }
        else
        {
            self.status.text = @"Error";
            
            if ( fault != nil )
            {
                UIAlertController *alert = [UIAlertController new];
                alert.title = @"Error";
                alert.message = fault.message;
                
                UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"Ok"
                                                                       style:UIAlertActionStyleCancel
                                                                     handler:^(UIAlertAction *action)
                                               {
                                                   
                                               }];
                
                [alert addAction:cancelAction];
                
                [self presentViewController:alert animated:YES completion:^
                 {
                     
                 }];
            }
        }
    }];
}


@end

File: Sample Apps/MedableSwift/MedableSwift/ViewController.swift
//
//  ViewController.swift
//  MedableSwift
//
//  Copyright (c) 2015 Medable Inc. All rights reserved.
//

import UIKit

class ViewController: UIViewController
{
    @IBOutlet var orgName : UIKit.UILabel?
    @IBOutlet var status : UIKit.UILabel?
    @IBOutlet var orgCode : UIKit.UILabel?
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        self.orgName?.text = ""
        self.orgCode?.text = ""
        
        self.status?.text = "Initialized"
    }

    @IBAction func getOrganization(_ sender: AnyObject)
    {
        let client = Medable.client()
        
        client.getPublicOrgInfo(
            callback: { (orgInfo, fault) -> Void in
                
                if let orgInfo = orgInfo
                {
                    self.orgName?.text = orgInfo["name"] as? String
                    self.orgCode?.text = orgInfo["code"] as? String
                    self.status?.text = "Ok"
                }
                else
                {
                    self.status?.text = "Error"
                    
                    if let fault = fault
                    {
                        let alert = UIAlertController()
                        alert.title = "Error"
                        alert.message = fault.message
                        
                        let cancelAction = UIAlertAction(title: "Ok", style: .cancel) { (action) in
                        }
                        alert.addAction(cancelAction)
                        
                        self.present(alert, animated: true, completion: { () -> Void in
                            
                        })
                    }
                }
            })
    }
}

